modules/idlgen/core/src/main/scala/avro/AvroSrcGenerator.scala

```diff
@@ -127,7 +127,8 @@ object AvroSrcGenerator extends SrcGenerator {
 
     val serviceLines =
       if (requestLines.isEmpty) Seq.empty
-      else Seq(s"@service trait ${protocol.getName}[F[_]] {", "") ++ requestLines :+ "}"
+      else
+        Seq(s"@service($serializationType) trait ${protocol.getName}[F[_]] {", "") ++ requestLines :+ "}"
```

Comment:
The `F[_]` parameter here, is it generated as a fresh symbol?

Reply:
This `AvroSrcGenerator` is a code generator, that generates a `*.scala` file from an `*.av[dl|pr]`, it doesn't do macros.